queries/clojure/highlights.scm

```diff
@@ -1,77 +1,333 @@
+  "release-pending-sends" "rem" "remove" "remove-all-methods"
+  "remove-method" "remove-ns" "remove-tap" "remove-watch"
+  "repeat" "repeatedly" "replace" "replicate"
+  "requiring-resolve" "reset!" "reset-meta!" "reset-vals!"
+  "resolve" "rest" "restart-agent" "resultset-seq" "reverse"
+  "reversible?" "rseq" "rsubseq" "run!" "satisfies?"
+  "second" "select-keys" "send" "send-off" "send-via"
+  "seq" "seq?" "seqable?" "seque" "sequence" "sequential?"
+  "set" "set-agent-send-executor!" "set-agent-send-off-executor!"
+  "set-error-handler!" "set-error-mode!" "set-validator!"
+  "set?" "short" "short-array" "shorts" "shuffle"
+  "shutdown-agents" "simple-ident?" "simple-keyword?"
+  "simple-symbol?" "slurp" "some" "some-fn" "some?"
+  "sort" "sort-by" "sorted-map" "sorted-map-by"
+  "sorted-set" "sorted-set-by" "sorted?" "special-symbol?"
+  "spit" "split-at" "split-with" "str" "string?"
+  "struct" "struct-map" "subs" "subseq" "subvec" "supers"
+  "swap!" "swap-vals!" "symbol" "symbol?" "tagged-literal"
+  "tagged-literal?" "take" "take-last" "take-nth" "take-while"
+  "tap>" "test" "the-ns" "thread-bound?" "to-array"
+  "to-array-2d" "trampoline" "transduce" "transient"
+  "tree-seq" "true?" "type" "unchecked-add" "unchecked-add-int"
+  "unchecked-byte" "unchecked-char" "unchecked-dec"
+  "unchecked-dec-int" "unchecked-divide-int" "unchecked-double"
+  "unchecked-float" "unchecked-inc" "unchecked-inc-int"
+  "unchecked-int" "unchecked-long" "unchecked-multiply"
+  "unchecked-multiply-int" "unchecked-negate" "unchecked-negate-int"
+  "unchecked-remainder-int" "unchecked-short" "unchecked-subtract"
+  "unchecked-subtract-int" "underive" "unquote"
+  "unquote-splicing" "unreduced" "unsigned-bit-shift-right"
+  "update" "update-in" "update-proxy" "uri?" "uuid?"
+  "val" "vals" "var-get" "var-set" "var?" "vary-meta" "vec"
+  "vector" "vector-of" "vector?" "volatile!" "volatile?"
+  "vreset!" "with-bindings*" "with-meta" "with-redefs-fn" "xml-seq"
+  "zero?" "zipmap"))
+
+
+
+;; >> Context based highlighting
 
-;; other calls
+; def-likes
+; Correctly highlight docstrings
 (list_lit
  .
- (sym_lit) @function)
+ (sym_lit) @a ; Don't really want to highlight twice
+ (#match? @a "^def.*")
+ .
+ (sym_lit)
+ .
+ (str_lit)? @text
+ .
+ (_))
 
-;; interop-ish
+; Funciton definitions
 (list_lit
  .
- (sym_lit) @method
- (#match? @method "^\\."))
-
-;; other symbols with dots
-((sym_lit) @variable
- (#match? @variable "\\."))
-
-;; quote
-(quoting_lit) @string.escape
+ (sym_lit) @a
+ (#any-of? @a "fn" "fn*" "defn" "defn-")
+ .
+ (sym_lit)? @function
+ .
+ (str_lit)? @text)
+;; TODO: Fix parameter highlighting
+;;       I think there's a bug here in nvim-treesitter
+;; TODO: Reproduce bug and file ticket
+ ;.
+ ;[(vec_lit
+ ;  (sym_lit)* @parameter)
+ ; (list_lit
+ ;  (vec_lit
+ ;   (sym_lit)* @parameter))])
+  
+;[((list_lit
+;   (vec_lit
+;    (sym_lit) @parameter)
+;   (_)
+;  +
+; ((vec_lit
+;   (sym_lit) @parameter)
+;  (_)))
```

Comment:
Yes, only some of the parameters are captured. I started to parser the parameters for common lisp directly in the parser. I could be related to a bug filed at Tree-Sitter that limited the number of returned results in starred expressions

Reply:
That sounds like the same issue I was having.
Only the first few parameters were being highlighted, even in very short parameter lists.